lib/rubygems/ext/rake_builder.rb

```diff
@@ -25,7 +25,7 @@ def self.build(extension, dest_path, results, args=[], lib_dir=nil, extension_di
       end
     end
 
-    rake_args = ["RUBYARCHDIR=#{dest_path}", "RUBYLIBDIR=#{dest_path}", *args]
+    rake_args = ["RUBYARCHDIR=#{dest_path}".shellescape, "RUBYLIBDIR=#{dest_path}".shellescape, *args]
```

Comment:
Are `args` safe enough to skip `shellescape` here?

Reply:
I'm not sure but note that previously the `run` method would `.join(" ")` its arguments (not `.shelljoin`) and pass them to `Open3.capture2e`. So if anything is broken after these changes, it was already broken before. The only reason I added `shellescape` here was to keep some tests that asserted for `RUBYARCHDIR\\=...` in error messages passing.

Reply:
Oooops. Since we don't go through the shell when spawning subcommands, it turns out we MUST NOT shellescape arguments! I'll fix this.

Reply:
I removed the shellescaping here and that unbroke things.